webook/static/modules/planner/dialog_manager/dialogManager.js

```diff
@@ -10,6 +10,7 @@
         this.onSubmit = onSubmit;
         this.onPreRefresh = onPreRefresh;
         this.dialogOptions = dialogOptions;
+        this._isRendering = false;
```

Comment:
I would like to know what this means. Is it just a flag to toggle `isOpen()`?

Reply:
It's a flag that indicates if rendering is happening.

The core issue here being that there's a very small period of time, where the dialog is rendering, that the render call can be fired again, causing the dialog to be rendered twice. So if you click an "Open Dialog" button twice in rapid succession, you can render the dialog twice. The intended behaviour is that if a dialog of some type is open, and a call for opening it is sent again, then the existing open dialog should be destroyed, and a new version rendered. 

So in practice; we lock the dialog, so that if it is actively being rendered then any subsequent calls for rendering it or opening it during the render will be ignored.

What we really do here, if I remember correctly is something akin to;

```py
if self._isRendering == False:
    self._isRendering = True # We are going to render, it may take some time for the htmlFabricator() method to finish.
    wait for render()
    self._isRendering = False # We are done rendering. This dialog may now be called for render again
```